libdnf/rpm/solv/goal_private.cpp

```diff
@@ -18,15 +18,38 @@ along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
 */
 
 #include "goal_private.hpp"
+#include "../../utils/utils_internal.hpp"
 
 extern "C" {
+#include <solv/evr.h>
 #include <solv/testcase.h>
 }
 
 namespace {
 
 
-void construct_job(libdnf::rpm::solv::IdQueue & job, libdnf::rpm::solv::IdQueue & install_only, bool force_best) {
+void allow_uninstall_all_but_protected(Pool * pool, libdnf::rpm::solv::IdQueue & job, const libdnf::rpm::solv::SolvMap * protected_packages, Id protected_kernel) {
+    libdnf::rpm::solv::SolvMap protected_pkgs(pool->nsolvables);
+    protected_pkgs.set_all();
```

Comment:
`protected_pkgs`? It is confusing. In fact, the meaning of the variable is the opposite. Please use another name, eg `not_protected_pkgs`.